test/solidity/simple-auction/simple-auction.js

```diff
@@ -0,0 +1,123 @@
+/*-
+ *
+ * Hedera Smart Contracts
+ *
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+const { expect } = require('chai')
+const { ethers } = require('hardhat')
+const Constants = require('../../constants')
+const Utils = require('../../hts-precompile/utils')
+
+const sleep = (ms) => new Promise((r) => setTimeout(r, ms));
+
+describe('Solidity Simple Auction example', function () {
+  let factory, signers, wallet, wallet1, 
+      contract, hasError, bidAmount,
+      contractShortLived, bidAmountSmall, initialEvent
+  const CONTRACT_DURATION = 10000000000
+  const CONTRACT_SHORT_DURATION = 1
+  const TRANSACTION_VALUE = '100'
+  const TRANSACTION_VALUE_SMALL = '10'
+
+  before(async function () {
+    signers = await ethers.getSigners()
+    wallet = signers[0]
+    wallet1 = signers[1]
+
+    factory = await ethers.getContractFactory(Constants.Contract.SimpleAuction)
+    contractShortLived = await factory.deploy(CONTRACT_SHORT_DURATION, wallet1.address)
+    await contractShortLived.deployed();
+
+    bidAmount = ethers.utils.parseUnits(TRANSACTION_VALUE, 'gwei')
+    bidAmountSmall = ethers.utils.parseUnits(TRANSACTION_VALUE_SMALL, 'gwei')
+    await sleep(5000)
```

Comment:
Q: why all these sleeps?

Reply:
I have two contracts deployed here. One is normal, the second is with a very short duration. After the next block it should be expired so I can test the expiration mechanism in the contract. 
That is why I am waiting for the next block to be mined before I start my tests. I am giving time for the `contractShortLived` to expire.

Reply:
Can we do this in a better way that sleeping.
Why not poll for a certain block. Sleeps are arbitrary and don't scale.
Poll for either a transaction or a block of desire then do the logic